src/jquery.eclipsefdn-api.js

```diff
@@ -1949,10 +1952,134 @@
     customFeaturedContent: function() {
       var $container = $($(this)[0].element);
       writeFeaturedContainer(this.settings.featuredContent, $container, this.settings.featuredContentType);
+    },
+
+    /**
+     Retrieves and inserts a list of promotions onto the page in the calling container. Will pass 
+     the publish target (if set) to target a particular sites promotions. These promotions will not
+     include or create impressions as they aren't meant to be consumed in production by the public.
+     */
+    allPromos: function() {
+      // set up the callbacks + containers for promos
+      var $container = $($(this)[0].element);
+      var self = this;
+      // create container for promos, to enable pagination bar
+      var $promosContainer = $container.find("> div.promos-container");
+      if ($promosContainer.length === 0) {
+        $promosContainer = $("<div></div>");
+        $promosContainer.attr({
+          "class": "promos-container",
+          "id": "promos-container-" + getPseudoRandomNumber()
+        });
+        $container.append($promosContainer);
+      }
+      if ($container.data("pagination") === true) {
+        $promosContainer.on("fetchPageItemsAd", retrievePromos);
+      }
+      // trigger first update of promos using first page
+      retrievePromosByPage($promosContainer, 1, 10);
+
+      /**
+       * Listener callback method for fetchPageItemsAd event.
+       */
+      function retrievePromos(event, page, pageSize) {
+        retrievePromosByPage(event.target, page, pageSize);
+      }
+
+      function retrievePromosByPage(contextEl, page, pageSize) {
+        var $currentContainer = $(contextEl);
+        var $parent = $currentContainer.parent();
+        var displayCount = $parent.data("count") || pageSize || 10;
+
+        // generate the URL for retrieve promotions
+        var url = self.settings.adsUrl;
+        var filter = "?page=" + page;
+        filter += "&pagesize=" + displayCount;
+        filter += convertDataToURLParameters($parent, "publish-target", "publish_to", undefined);
+        
+        console.log(url + filter);
```

Comment:
We should remove this console for final version

Reply:
Done